core/src/main/scala/better/files/File.scala

```diff
@@ -690,7 +690,6 @@ class File private(val path: Path)(implicit val fileSystem: FileSystem = path.ge
     */
   def copyTo(destination: File, overwrite: Boolean = false)(implicit copyOptions: File.CopyOptions = File.CopyOptions(overwrite)): destination.type = {
     if (isDirectory) {//TODO: maxDepth?
-      if (overwrite) destination.delete(swallowIOExceptions = true)
```

Comment:
Based on original issue(dir to dir with overwrite), it did fix the issue. The minimal test cases would be 3 test cases(dir -> dir, file -> file, and file -> fir). It's wired to have dir to file, in *nix environment it wouldn't work. What I patched to [FileSpec.scala](https://github.com/pathikrit/better-files/pull/164/files#diff-3c136f71bba4f951c2bd8d8e781346d6) is to verify dir to dir with overwrite. File to file was already to test in [L335](https://github.com/pathikrit/better-files/blob/master/core/src/test/scala/better/files/FileSpec.scala#L335) of FileSpec.scala. I did found file to dir would not work, it would throw `java.nio.file.DirectoryNotEmptyException`. But whether or not I remove this line, it would happen. As I investigated, I found `java.nio.file.Files.copy` which `copyTo` based on doesn't support this behavior. I thought **file to dir** could open another issue to see how's going.